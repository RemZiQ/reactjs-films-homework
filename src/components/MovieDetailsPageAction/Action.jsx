import React, { Component } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';

import { getTrailer, noError } from '../../modules/module_Search/actions';
import Trailer from '../MovieListItem/Trailer';
import TrailerError from '../MovieListItem/TrailerError';
import Iframe from '../MovieListItem/Trailer/Iframe';
import './action.scss';

class Action extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showInfo: false,
      showTrailer: false,
    };
  }

  toogleInfo = () => {
    const { showInfo } = this.state;
    this.setState({ showInfo: !showInfo });
  };

  toogleTrailer = () => {
    const { showTrailer } = this.state;
    this.setState({ showTrailer: !showTrailer });
  }

  showTrailer = () => {
    const { fetchTrailer, setToNoError } = this.props;
    if (this.props.currentMovie) {
      const { id } = this.props;
      fetchTrailer(id);
    }
    this.toogleTrailer();
  }

  // handle = (e) => {
  //   handleShowTrailer();
  //   try {
  //     fetchTrailer(e);
  //   } catch (error) {
  //     console.log(error);
  //   }
  // };

  render() {
    const { info } = this.props;
    const { showInfo, showTrailer } = this.state;
    const infoToggled = showInfo ? <div className="action_info">{info}</div> : null;
    // must be added trailer
    if (showTrailer) {
      return (
        <div className="action_container">
          {infoToggled}
          <div className="action__buttons_container">
            <button onClick={this.showTrailer} type="button" className="action__watch">Watch now</button>
            {console.log(this.state.showTrailer)}
            <button onClick={this.toogleInfo} type="button" className="action__view">View info</button>
            <Trailer>
              <div className="Trailer">
                <button onClick={this.toogleTrailer} type="button" id={123} className="movieList__item__trailer_closeButton">
                  <i className="fas fa-times" />
                </button>
                <TrailerError>
                  <Iframe trailer={'something'} error={false} />
                </TrailerError>
              </div>
            </Trailer>
          </div>
        </div>
      );
    }
    return (
      <div className="action_container">
        {infoToggled}
        <div className="action__buttons_container">
          <button onClick={this.toogleTrailer} type="button" className="action__watch">Watch now</button>
          {console.log(this.state.showTrailer)}
          <button onClick={this.toogleInfo} type="button" className="action__view">View info</button>
        </div>
      </div>
    );
  }
}

Action.propTypes = {
  info: propTypes.string,
};

Action.defaultProps = {
  info: 'Sorry, no information here',
};

const mapStateToProps = store => ({ currentMovie: store.currentMovie });
const mapDispatchToProps = dispatch => ({
  fetchTrailer: (id) => {
    dispatch(getTrailer(id));
  },
  setToNoError: () => dispatch(noError()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Action);
