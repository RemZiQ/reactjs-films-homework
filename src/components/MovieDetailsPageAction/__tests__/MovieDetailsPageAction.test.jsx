import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';
import { act, Simulate } from 'react-dom/test-utils';
import Action from '../Action';

let container;
let modal;

beforeEach(() => {
  container = document.createElement('div');
  container.setAttribute('id', 'modal');
  modal = document.createElement('div');
  modal.setAttribute('id', 'modal');
  document.body.appendChild(container);
  document.body.appendChild(modal);
});

afterEach(() => {
  document.body.removeChild(container);
  document.body.removeChild(modal);
  container = null;
  modal = null;
});

it('StarRating renders well without and with state { showTrailer: true }', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<Action />);
  expect(result).toMatchSnapshot();
});

// it('example 3.0', () => {
//   act(() => {
//     const result = TestRenderer.create(<Action />, container);
//     console.log('4321', document.getElementById('modal').id);
//     // const instance = result.getInstance();
//     // const toogleTrailer = instance.toogleTrailer;
//     // toogleTrailer();
//     // expect(result.toJSON).toMatchSnapshot();
//   });
// });

