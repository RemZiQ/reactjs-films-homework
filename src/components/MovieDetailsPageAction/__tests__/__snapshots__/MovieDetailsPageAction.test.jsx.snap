// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating renders well 1`] = `
<div
  className="action_container"
>
  <div
    className="action__buttons_container"
  >
    <button
      className="action__watch"
      onClick={[Function]}
      type="button"
    >
      Watch now
    </button>
    <button
      className="action__view"
      onClick={[Function]}
      type="button"
    >
      View info
    </button>
  </div>
</div>
`;

exports[`StarRating renders well without and with state { showTrailer: true } 1`] = `
<div
  className="action_container"
>
  <div
    className="action__buttons_container"
  >
    <button
      className="action__watch"
      onClick={[Function]}
      type="button"
    >
      Watch now
    </button>
    <button
      className="action__view"
      onClick={[Function]}
      type="button"
    >
      View info
    </button>
  </div>
</div>
`;

exports[`StarRating renders well without and with state { showTrailer: true } 2`] = `
<div
  className="action_container"
>
  <div
    className="action__buttons_container"
  >
    <button
      className="action__watch"
      onClick={[Function]}
      type="button"
    >
      Watch now
    </button>
    <button
      className="action__view"
      onClick={[Function]}
      type="button"
    >
      View info
    </button>
    <Trailer>
      <div
        className="Trailer"
      >
        <button
          className="movieList__item__trailer_closeButton"
          id={123}
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-times"
          />
        </button>
        <TrailerError>
          <Iframe
            trailer="https://www.youtube.com/embed/kVrqfYjkTdQ&t"
          />
        </TrailerError>
      </div>
    </Trailer>
  </div>
</div>
`;

exports[`StarRating renders well2222 1`] = `
<div
  className="action_container"
>
  <div
    className="action__buttons_container"
  >
    <button
      className="action__watch"
      onClick={[Function]}
      type="button"
    >
      Watch now
    </button>
    <button
      className="action__view"
      onClick={[Function]}
      type="button"
    >
      View info
    </button>
    <Trailer>
      <div
        className="Trailer"
      >
        <button
          className="movieList__item__trailer_closeButton"
          id={123}
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-times"
          />
        </button>
        <TrailerError>
          <Iframe
            trailer="https://www.youtube.com/embed/kVrqfYjkTdQ&t"
          />
        </TrailerError>
      </div>
    </Trailer>
  </div>
</div>
`;
